test/gtest/mhp/slide_view-3.cpp

```diff
@@ -13,6 +13,23 @@ TYPED_TEST(Slide3, suite_works_for_3_processes_only) {
   EXPECT_EQ(dr::mhp::default_comm().size(), 3); // dr-style ignore
 }
 
+TYPED_TEST(Slide3, different_halos) {
+  TypeParam dv(6, dr::mhp::distribution().halo(1, 2));
+  iota(dv, 1);
+  dv.halo().exchange();
+}
+
+TYPED_TEST(Slide3, different_halos_with_0) {
+  TypeParam dv1(6, dr::mhp::distribution().halo(3, 0));
+  iota(dv1, 1);
+  dv1.halo().exchange();
```

Comment:
Could you please add some checks after exchange (here and in other tests)? Otherwise iota call in this test is not needed. E.g write something to halo region of second process, do exchange and check if new value appeared in the first (or the third) process?